src/components/FilterBar/FilterDisplay/Inputs/RelativeDateRangeInput.react.js

```diff
@@ -0,0 +1,208 @@
+import { RangeInput } from "./RangeInput.react";
+import { DateInput } from "./DateInput.react";
+export class RelativeDateRangeInput extends RangeInput {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: this.props.value || { from: null, to: null },
+      operator: this.props.value.operator || "absolute"
+    }
+  }
+
+  isAbsolute() {
+    return this.state.operator == "absolute"
+  }
+
+  intToMoment(value) {
+    if (Number.isNaN(value) || value == null) {
+      return null
+    }
+
+    if (value < 0) {
+      return moment().subtract(Math.abs(value), 'day')
+    } else {
+      return moment().add(Math.abs(value), 'day')
+    }
+  }
+
+  momentFormatted(value) {
+    var momentValue = this.intToMoment(parseInt(value))
+
+    if (momentValue) {
+      return momentValue.format(this.props.dateFormat)
+    } else {
+      return moment().format(this.props.dateFormat)
```

Comment:
So if we can't parse the value, we're just returning the current date, right?

Reply:
Yes. This is to display the numbers as actual dates next to the filter. `momentValue` will be null for 0 or NaN inputs which always means today's date.